libraries/exoplayer/src/main/java/androidx/media3/exoplayer/upstream/Loader.java

```diff
@@ -90,6 +90,18 @@ public interface Loadable {
   /** A callback to be notified of {@link Loader} events. */
   public interface Callback<T extends Loadable> {
 
+    /**
+     * Called when a load has started for the first time or through a retry.
+     *
+     * @param loadable The loadable whose load has completed.
+     * @param elapsedRealtimeMs {@link SystemClock#elapsedRealtime} when the load ended.
+     * @param loadDurationMs The duration in milliseconds of the load since {@link #startLoading}
+     *     was called.
+     * @param error The load error of the previous load attempt if this is a retry, otherwise null
```

Comment:
`error` and `errorCount` are already passed to `onLoadError` - which I think means an implementation of this interface already has access to this information without needing to pass it here too?

Reply:
Ah good point, we can cache that value on the impl side. Ultimately, I wanted a way to signify that the onLoadStart attempt was a retry or first attempt but agreed it can be tracked internally

Reply:
After some discussion on the internal review, we're leaning towards putting some sort of parameter back in here to (as you suggested) make it clear that the meaning of 'start' here is distinct from the meaning in `startLoading` (i.e. this will be called once per retry).

So I'm going to put back your `errorCount` parameter, but rename it `retryCount`.

We're not planning to include the `error` parameter, because as agreed that's available from the `onLoadError` method.